frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -161,7 +155,6 @@ export class UnconnectedDataSelector extends Component {
     useOnlyAvailableSchema: PropTypes.bool,
     isInitiallyOpen: PropTypes.bool,
     tableFilter: PropTypes.func,
-    hasTableSearch: PropTypes.bool,
```

Comment:
`hasTableSearch` was always `false` (except in unit tests) so I removed it.
Other code removals in this PR are a consequence of this change.